merfi/collector.py

```diff
@@ -66,3 +71,36 @@ def debian_release_files(self):
                     if isfile(release_file):
                         result.append(release_file)
         return result
+
+    @property
+    def rpm_repomd_files(self):
```

Comment:
this and `rpm_files` look very close to what we had initially with this file collector where we could pass a regular expression. I understand that this depends on the fact that it needs to be an RPM repo though, not just any `repomd.xml` file.

A "repository collector" doesn't sound like something that would "collect rpm files". What do you think if this logic gets stripped out into a "File Collector", something less fancy that what we had initially but one that we can re-use to get both repomd files, rpm files, and whatever comes next?

It might be time too to implement something like an `RpmRepo` class that with the help of the collector would allow us to have a nice syntax like:

    >>> repo = RpmRepo('/opt/repo/rpms')
    >>> print repo.rpms
    [</opt/repo/rpms/ceph-1.0.0.rpm>, </opt/repo/rpms/ceph-1.0.1.rpm>]
    >>> for rpm in repo.rpms:
        print rpm.path, rpm.name
    ...
    /opt/repo/rpms/ceph-1.0.0.rpm ceph-1.0.0.rpm
    /opt/repo/rpms/ceph-1.0.1.rpm ceph-1.0.1.rpm

    >>> print repo.repomd
    ["/opt/repo/rpms/repodata/repomd.xml"]

This is just toying around, but would certainly take us out of rpm-specific logic into this collector that intended to be generic.